vtysh/vtysh.c

```diff
@@ -652,8 +677,16 @@ static int vtysh_execute_func(const char *line, int pager)
 		if (cmd_stat != CMD_SUCCESS)
 			break;
 
-		if (cmd->func)
-			(*cmd->func)(cmd, vty, 0, NULL);
+		if (cmd->func) {
+			/* XX
+			 * Need make it better
+			 */
+			if (!strcmp(cmd->string, "hostname WORD")
+			    || !strcmp(cmd->string, "doaminname WORD"))
+				(*cmd->func)(cmd, vty, 1, (struct cmd_token **)line);
```

Comment:
Sorry but this is a hard NACK. The last thing vtysh needs is more special casing.

Also, you misspelled `domainname` which tells me that whatever this does, it isn't tested.

Reply:
Sorry for the misspelled word, i append a commit to fix it. It is caused by my git operation of combining many to one commit.
On the contrary, i do much test.

Yes, this hard code here is ugly. I have no idea of it, i will try other method. And can you shed light on it?

Reply:
Since vtysh send commands to other daemons, then other daemons can call hostname_cmd.
vtysh need also run these commands (call hostname_cmd or domainname_cmd), here is for that.